src/platform/datapath_winrdma.c

```diff
@@ -0,0 +1,5151 @@
+    }
+
+    Status = NdspiCreateOverlappedFile(RdmaAdapter, &RdmaAdapter->OverlappedFile);
+    if (QUIC_FAILED(Status)) {
+        QuicTraceLogError(
+            CreateOverlappedFile,
+            "CreateAdapterOverlappedFile failed, status: %d", Status);
+        goto ErrorExit;
+    }
+
+    CxPlatPoolInitializeEx(
+        FALSE,
+        sizeof(RDMA_CONNECTION),
+        QUIC_POOL_SOCKET,
+        MAX_RDMA_CONNECTION_POOL_SIZE,
+        NULL,
+        NULL,
+        &RdmaAdapter->ConnectionPool);
+    
+    CxPlatPoolInitializeEx(
+        FALSE,
+        sizeof(RDMA_SEND_RING_BUFFER),
+        QUIC_POOL_SOCKET,
+        MAX_RDMA_CONNECTION_POOL_SIZE,
+        NULL,
+        NULL,
+        &RdmaAdapter->SendRingBufferPool);
+
+    CxPlatPoolInitializeEx(
+        FALSE,
+        sizeof(RDMA_RECV_RING_BUFFER),
+        QUIC_POOL_SOCKET,
+        MAX_RDMA_CONNECTION_POOL_SIZE,
+        NULL,
+        NULL,
+        &RdmaAdapter->RecvRingBufferPool);
+
+    CxPlatPoolInitializeEx(
+        FALSE,
+        sizeof(RDMA_RECV_RING_BUFFER),
+        QUIC_POOL_SOCKET,
+        MAX_RDMA_CONNECTION_POOL_SIZE,
+        NULL,
+        NULL,
+        &RdmaAdapter->RemoteRingBufferPool);
+
+    //
+    // Populate the Adapter Info to get MAX values supported
+    // for the adapter
+    //
+    ULONG AdapterInfoSize = sizeof(RdmaAdapter->AdapterInfo);
+    Status = RdmaAdapter->Adapter->lpVtbl->Query(
+        RdmaAdapter->Adapter,
+        &RdmaAdapter->AdapterInfo,
+        &AdapterInfoSize);
+    if (QUIC_FAILED(Status)) {
+        QuicTraceLogError(
+            QueryAdapterInfoFailed,
+            "QueryAdapterInfo failed, status: %d", Status);
+        goto ErrorExit;
+    }
+
+    return Status;
+
+ErrorExit:
+
+    CxPlatRdmaAdapterRelease(RdmaAdapter);
+
+    return Status;
+}
+
+//
+// Cleanup an RDMA context
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+CxPlatRdmaAdapterRelease(
+    _In_ void* Adapter)
+{
+    RDMA_NDSPI_ADAPTER* NdAdapter = (RDMA_NDSPI_ADAPTER*) Adapter;
+    if (NdAdapter)
+    {
+        if (NdAdapter->Adapter)
+        {
+            NdAdapter->Adapter->lpVtbl->Release(NdAdapter->Adapter);
+        }
+
+        if (NdAdapter->OverlappedFile)
+        {
+            CloseHandle(NdAdapter->OverlappedFile);
+        }
+
+        CxPlatPoolUninitialize(&NdAdapter->ConnectionPool);
+        CxPlatPoolUninitialize(&NdAdapter->SendRingBufferPool);
+        CxPlatPoolUninitialize(&NdAdapter->RecvRingBufferPool);
+        CxPlatPoolUninitialize(&NdAdapter->RemoteRingBufferPool);
+
+        CXPLAT_FREE(NdAdapter, QUIC_POOL_DATAPATH);
+    }
+}
+
+//
+// Free Child Objects allocated within an RDMA_CONNECTION object
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+RdmaConnectionFree(
+    _In_ RDMA_CONNECTION* RdmaConnection
+)
+{
+    if (!RdmaConnection)
+    {
+        return;
+    }
+    
+    if (RdmaConnection->QueuePair)
+    {
+        RdmaConnection->QueuePair->lpVtbl->Release(RdmaConnection->QueuePair);
```

Comment:
By curiosity, what is your logic for function that your wrapped vs these direct calls?